cmd/snap/cmd_whoami.go

```diff
@@ -34,12 +40,47 @@ The whoami command shows the email the user is logged in with.
 
 type cmdWhoAmI struct {
 	clientMixin
+
+	// XXX: rename to --{resync,overwrite,rewrite} or similar?
+	RefreshSSHKeys bool `long:"refresh-ssh-keys"`
 }
 
 func init() {
 	addCommand("whoami", shortWhoAmIHelp, longWhoAmIHelp, func() flags.Commander { return &cmdWhoAmI{} }, nil, nil)
 }
 
+func refreshSSHKeys(email string) error {
+	if release.OnClassic {
+		return fmt.Errorf("cannot refresh keys: not supported on classic")
+	}
+	if email == "" {
+		return fmt.Errorf("cannot refresh keys: no email")
+	}
+
+	// FIXME: set auth context
+	var storeCtx store.DeviceAndAuthContext
+	sto := storeNew(nil, storeCtx)
+	userInfo, err := sto.UserInfo(email)
+	if err != nil {
+		return fmt.Errorf("cannot refresh keys for %q: %s", email, err)
+	}
+	if len(userInfo.SSHKeys) == 0 {
+		return fmt.Errorf("cannot reresh keys for %q: no ssh keys found", email)
```

Comment:
typo: refresh

Though maybe it's better to remove all these "cannot refresh keys for %q" prefixes from the error message, and add them in the place where this function is called? Not sure if there's a convention in the golang community about how error messages should be formatted.

Reply:
I tweak the code a bit more